src/edge/tests/test_ssr.py

```diff
@@ -1017,6 +1182,317 @@ def allowed():
         )
 
 
+class IntegrationRecombinationAnnotationTest(TestCase):
+
+    def test_can_integrate_multiple_places_on_fragment_with_annotations(self):
+        parent_genome = Genome(name="foo")
+        parent_genome.save()
+        f = Fragment.create_with_sequence("bar", "attg" + "tttt" + "attg")
+
+        Genome_Fragment(genome=parent_genome, fragment=f, inherited=False).save()
+
+        class FakeReaction(Reaction):
+            @staticmethod
+            def allowed():
+                return [Integration("attg", "attg", "attc", "atcc")]
+
+        r = FakeReaction(parent_genome, "ggg" + "attg" + "a" * 100, True)
+        annotations = [
+            {
+                "base_first": 4, "base_last": 7, "feature_name": "replaced",
+                "feature_type": "gene", "feature_strand": 1
+            },
+            {
+                "base_first": 3, "base_last": 8, "feature_name": "replaced overlap",
+                "feature_type": "cds", "feature_strand": -1
+            },
+            {
+                "base_first": 10, "base_last": 50, "feature_name": "test gene",
+                "feature_type": "gene", "feature_strand": 1
+            },
+            {
+                "base_first": 60, "base_last": 100, "feature_name": "test cds",
+                "feature_type": "cds", "feature_strand": -1
+            },
+        ]
+        child_genome = r.run_reaction("far", annotations=annotations)
+        f = child_genome.fragments.all()[0].indexed_fragment()
+
+        anns = f.annotations()
+        self.assertEqual(len(anns), 4)
+
+        self.assertEqual(anns[0].base_first, 7)
```

Comment:
can you add a comment re if anns[0].base_first is 0-indexed or 1-indexed? If it's 1-indexed, should be 8, no?